src/mame/drivers/sms_bootleg.cpp

```diff
@@ -272,34 +435,169 @@ DRIVER_INIT_MEMBER(smsbootleg_state,sms_supergame)
 	{
 		rom[i] ^= 0x80;
 	}
+
+	for (int i = 0;i < 0x200;i++)
+	{
+		m_unpaged->configure_entry(i, &m_mainrom[(i & 0x1fe) * 0x4000]);
+		m_page0->configure_entry(i, &m_mainrom[0x400 + i * 0x4000]);
+		m_page1->configure_entry(i, &m_mainrom[i * 0x4000]);
+		m_page2->configure_entry(i, &m_mainrom[i * 0x4000]);
```

Comment:
You should be able to avoid the loop with `configure_entrees`.  Also consider that you always apply a shift when setting the bank for `m_unpaged` - you can avoid that as well by taking it into consideration when configuring banks.  Try something like this for configuring banks:
```
m_unpaged->configure_entry(0, 0x100, &m_mainrom[0], 0x8000]);
m_page0->configure_entries(0, 0x200, &m_mainrom[0x400], 0x4000]);
m_page1->configure_entries(0, 0x200, &m_mainrom[0], 0x4000]);
m_page2->configure_entries(0, 0x200, &m_mainrom[0], 0x4000]);
```

Then you can simplify the line for setting the `m_unpaged` bank to get rid of the shift:
```
m_unpaged->set_entry(m_bankbase);
```